### PR TITLE
Mobility episode generator and RearrangeSim modification

### PR DESCRIPTION
**Pull request by Tradewindycc**
## Motivation and Context
Add the mobility_episode_generator for mp3d episode, and adapt RearrangeSim for MP3D dataset

## How Has This Been Tested

1. Download MP3D-habitat dataset by download_mp.py, the dataset path will be like habitat-lab/data/scene_datasets/mp3d/...
2. Download mobility_eval.json.gz for mobility eval from [here](https://drive.google.com/drive/folders/1q_Khj1JpDE0zm4jv1WG1Mn3hlPydvMTc?usp=drive_link), and put it into data/datasets/mobility like habitat-lab/data/datasets/mobility/mobility_eval.json.gz. Or you can generate mobility episodes yourself by mobility_episodes_generator.py, remember to modify the config path in it.
3. run the habitat_baselines.run with config-name=social_rearrange/llm_spot_fetch_mobility.yaml

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation
- **\[Refactoring\]** Large changes to the code that improve its functionality or performance
- **\[Dependency Upgrade\]** Upgrades one or several dependencies in habitat
- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!
- **\[Experiment\]** A pull request that add new features to the [habitat-baselines](/habitat-baselines/) training codebase. Experiments Pull Requests can be any size, must have smoke/integration tests and be isolated from the rest of the code. Your code additions should not rely on other habitat-baselines code. This is to avoid dependencies between different parts of habitat-baselines. You must also include a README that will document how to use your new feature or trainer. **You** will be the maintainer of this code, if the code becomes stale or is not supported often enough, we will eventually remove it.

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: download_mp.py
Comment (SgtVincent) [inline download_mp.py:1]: We cannot incorporate this file to the repo since this download script can only be LICENSED BY AUTHOR.

File: habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py
Comment (SgtVincent) [inline habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py]: why do you commented out this line?

File: habitat-baselines/habitat_baselines/rl/multi_agent/multi_llm_policy.py
Comment (SgtVincent) [inline habitat-baselines/habitat_baselines/rl/multi_agent/multi_llm_policy.py:1]: Please review the PR before committing. We'd better keep these temp modifications locally

File: habitat-lab/habitat/tasks/nav/nav.py
Comment (SgtVincent) [inline habitat-lab/habitat/tasks/nav/nav.py]: Please stick to the rule of minimum change. Random changes like this should be avoided.

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py
Comment (SgtVincent) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py]: Can we wrap this snippet of code to a function in utils, for example? I don't want to dilute the code in core classes. 
If this requires a lot of effort, then simply ignore my comment here. :)

File: habitat-lab/habitat/tasks/rearrange/rearrange_task.py
Comment (SgtVincent) [inline habitat-lab/habitat/tasks/rearrange/rearrange_task.py]: Can we revert this change?

File: habitat-lab/habitat/utils/visualizations/maps.py
Comment (SgtVincent) [inline habitat-lab/habitat/utils/visualizations/maps.py]: why do we want to upgrade imageio version to v2?

**Review by SgtVincent — CHANGES_REQUESTED** [review]
I have some questions and suggestions. Please drop me message when you want a second review.

File: habitat-lab/habitat/tasks/rearrange/rearrange_task.py
Comment (Tradewindycc) [reply habitat-lab/habitat/tasks/rearrange/rearrange_task.py]: Got it. Done.

File: download_mp.py
Comment (Tradewindycc) [reply download_mp.py:1]: oops! Sorry for that.

File: habitat-lab/habitat/tasks/nav/nav.py
Comment (Tradewindycc) [reply habitat-lab/habitat/tasks/nav/nav.py]: Got it. This change has been reverted.

File: habitat-lab/habitat/utils/visualizations/maps.py
Comment (Tradewindycc) [reply habitat-lab/habitat/utils/visualizations/maps.py]: There are some warnings when running the code, forgot to revert.

File: habitat-baselines/habitat_baselines/rl/multi_agent/multi_llm_policy.py
Comment (Tradewindycc) [reply habitat-baselines/habitat_baselines/rl/multi_agent/multi_llm_policy.py:1]: Understood.

File: habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py
Comment (Tradewindycc) [reply habitat-baselines/habitat_baselines/rl/hrl/hierarchical_policy.py]: forgot to revert this change when committing.  o.o

File: habitat-baselines/habitat_baselines/rl/multi_agent/multi_llm_policy.py
Comment (Tradewindycc) [reply habitat-baselines/habitat_baselines/rl/multi_agent/multi_llm_policy.py:1]: Still, RearrangeSim initialized with MP3D scene can't get env_context_text through hssd sensor.

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py
Comment (Tradewindycc) [reply habitat-lab/habitat/tasks/rearrange/rearrange_sim.py]: Cross-reference will happen after wrapping this snippet into utils.py, I wrapped it into a class function instead.

**Review by SgtVincent — APPROVED** [review]
LGTM!

**Pull request merged**
